src/app/projects/services/doi.service.ts

```diff
@@ -0,0 +1,138 @@
+
+  loading: Subject<boolean> = new Subject<boolean>();
+
+  constructor(private http: HttpClient,
+              private ingestService: IngestService,
+              private alertService: AlertService,
+              private router: Router) {
+  }
+
+  importProjectUsingDoi(doi: string) {
+    this.loading.next(true);
+    forkJoin({
+      projects: this.getProjectsWithDOI(doi),
+      doiExists: this.doesDoiExist(doi)
+    }).subscribe(({projects, doiExists}) => {
+        this.showExistingProjectsAlert(projects, 'doi');
+
+        if (!doiExists) {
+          this.showDOINotExistsAlert(doi);
+        }
+
+        if (doiExists && projects.length == 0) {
+          this.createProjectWithDoi(doi);
+        }
+
+        this.loading.next(false)
+      },
+      error => {
+        this.alertService.error('An error occurred', error.message);
+        this.loading.next(false);
+      }
+    );
+  }
+
+  getProjectDetails(searchUsing: Identifier, searchString: string): Observable<AutofillProject> {
+    return this.queryEuropePMC(searchUsing, searchString).pipe(
+      map(response => {
+        const result = response.resultList.result[0];
+        if (result) {
+          return this.createAutoFillProject(result);
+        } else {
+          throw throwError(response);
+        }
+      })
+    );
+  }
+  
+  private showExistingProjectsAlert(projects: Project[], publicationIdType: string) {
+    projects.forEach(project => {
+      const projectTitle = project?.content?.['project_core']?.['project_title'];
+      const link = `/projects/detail?uuid=${project?.uuid?.uuid}`;
+      const errorTitle = `This ${publicationIdType} has already been used by project:`
+      this.alertService.error(
+        errorTitle,
+        `<a href="${link}">${projectTitle}</a>`);
+    });
+  }
+
+  private getProjectsWithDOI(doi: string): Observable<Project[]> {
+    const criteria = {
+      'field': 'content.publications.doi',
+      'operator': 'IS',
+      'value': doi
+    };
+    return this.ingestService.getProjectsUsingCriteria(criteria);
+  }
+
+  private doesDoiExist(doi: string): Observable<boolean> {
+    const searchIdentifier = Identifier.DOI;
+    return this
+      .queryEuropePMC(searchIdentifier, doi)
+      .pipe(
+        map(response => response.resultList.result.length > 0)
+      );
+  }
+
+  private queryEuropePMC(queryId: string, queryString: string): Observable<EuropePMCHttpSearchResponse> {
+    const params = {
+      query: queryId + ':' + queryString,
+      resultType: 'core',
+      format: 'json'
+    };
+    return this.http.get<EuropePMCHttpSearchResponse>(this.API_URL, {params});
+  }
+
+  private createProjectWithDoi(doi) {
+    const params = {
+      [Identifier.DOI]: doi
+    };
+    this.router.navigate(['/projects', 'register'], {queryParams: params});
+  }
+
+  private showDOINotExistsAlert(doi) {
+    const link = `mailto:wrangler-team@data.humancellatlas.org?subject=Cannot%20find%20project%20by%20DOI&body=${doi}`;
```

Comment:
please make email address, subject, etc configurable.

Reply:
I've made the email address configurable, but not the subject and message. It's very simple atm and specific to this error scenario, I don't think there's a need.